### PR TITLE
ARROW-2743: [Java] Change CI script to remove project artifacts before building

### PR DESCRIPTION
**Pull request by BryanCutler**
This adds a command to the Java CI script to remove project artifacts, of the current version, from the local Maven repo prior to building.  This is to catch a potential issue where the build might use parent poms from the local repo instead of in the current project. 

The history of this issue is ARROW-1780 added a child pom 2 levels deep that used the arrow-java-root as parent but did not specify the parent path.  The build was able to pass because it found the parent pom in the local repo, but should have been using the one from the project.  Bug ARROW-2727 was reported because when trying a clean build without the parent pom in the repo, the build will fail.  This was resolved by specifying the correct parent relative path.

### CODE REVIEW

**Comment by BryanCutler** [comment]
I added a fix that removes the project artifacts from the local repo, then I reverted the previous fix with the relative path.  So this should cause CI to fail.

File: java/adapter/jdbc/pom.xml [hunk 1]
```diff
@@ -17,7 +17,6 @@
         <groupId>org.apache.arrow</groupId>
         <artifactId>arrow-java-root</artifactId>
         <version>0.10.0-SNAPSHOT</version>
-        <relativePath>../../pom.xml</relativePath>
```
Comment (BryanCutler) [inline java/adapter/jdbc/pom.xml]: This change is to hopefully cause the failure

**Comment by BryanCutler** [comment]
I could not get CI to fail.  From what I can tell, the issue does not happen with a newer version of maven.  It must be able to locate the parent POM some other way, and it is not installed in the local repo.

I verified this locally by removing the parent POM relative path and running the cmd:
`mvn -B install build-helper:remove-project-artifact -Dbuildhelper.failOnError=false`

With mvn 3.3.9, I get the error saying it can not find the arrow-java-root POM, but with mvn 3.5.2 (the version used by CI) it is able find it and completes without error. I'm not sure there is anything we can do to catch a problem like this in the future except move to a lower version of mvn, what are your thoughts @wesm ?

**Comment by wesm** [comment]
I think it's OK to merge a fix even if we can't trigger the failure in Travis CI (since it seems it's relying on some kind of caching behavior anyway).

Can the same effect be reaped by deleting a particular directory under `$HOME/.m2`?

**Comment by BryanCutler** [comment]
@wesm from what I can tell, Travis doesn't have anything cached in the local maven repo under `/home/travis/.m2/repository`, so this fix here would do nothing.  I think something changed in the newer versions of Maven so that it can find the parent pom within the same project without needing a relative path.  I can't find any documentation on this, but I'm able to see this behavior locally with Maven 3.5.2.  I'm going to close this since this change doesn't work and I'm out of ideas for now.

**Pull request closed without merging**